pkg/tsdb/interval/interval.go

```diff
@@ -135,7 +135,7 @@ func FormatDuration(inter time.Duration) string {
 }
 
 //nolint: gocyclo
-func roundInterval(interval time.Duration) time.Duration {
+func RoundInterval(interval time.Duration) time.Duration {
```

Comment:
We might want to extract all/parts of the interval calculation to the SDK eventually. But as a first step might create a new calculator that operates on SDK constracts?

Reply:
We are already using that on this PR, no?

Reply:
I think you'll realize this is needed if you replace the usage of `models.DataSource` in the client package :) `GetIntervalFrom` is used there for example.

Reply:
Added some comments in the new file. Guess the changes in this file can be reverted then?